nsq/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/NsqSink.scala

```diff
@@ -19,28 +19,80 @@
 package com.snowplowanalytics.snowplow.collectors.scalastream
 package sinks
 
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeoutException
+
 import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContextExecutorService
+import scala.concurrent.duration.MILLISECONDS
+
+import cats.effect.{Resource, Sync}
 
 import com.snowplowanalytics.client.nsq.NSQProducer
-import com.snowplowanalytics.snowplow.collectors.scalastream.model._
+import com.snowplowanalytics.snowplow.collector.core.{Sink}
+import com.snowplowanalytics.client.nsq.exceptions.NSQException
 
 /**
   * NSQ Sink for the Scala Stream Collector
   * @param nsqConfig Configuration for Nsq
   * @param topicName Nsq topic name
   */
-class NsqSink(val maxBytes: Int, nsqConfig: Nsq, topicName: String) extends Sink {
+class NsqSink[F[_]: Sync] private (
+  val maxBytes: Int,
+  nsqConfig: NsqSinkConfig,
+  topicName: String,
+  executorService: ScheduledExecutorService
+) extends Sink[F] {
+
+  implicit lazy val ec: ExecutionContextExecutorService =
+    concurrent.ExecutionContext.fromExecutorService(executorService)
+
+  var healthStatus = true
+
+  override def isHealthy: F[Boolean] = Sync[F].pure(healthStatus) 
 
   private val producer = new NSQProducer().addAddress(nsqConfig.host, nsqConfig.port).start()
 
+  def produceData(topicName: String, events: List[Array[Byte]]): Unit =
+    try {
+      producer.produceMulti(topicName, events.asJava)
+    } catch {
+      case e @ (_: NSQException | _: TimeoutException) => {
+        healthStatus = false
+        throw e
+      }
+    }
```

Comment:
```suggestion
def produceData(topicName: String, events: List[Array[Byte]]): F[Unit] =
    Sync[F].blocking(producer.produceMulti(topicName, events.asJava))
      .onError { case _: NSQException | _: TimeoutException =>
        Sync[F].delay(healthStatus = false)
      } *> Sync[F].delay(healthStatus = true)
```
Requires to `import cats.implicits._`

Health status was never made healthy again.

Reply:
But without retries a failure will crash the app so there isn't much point in setting the health status to `false` it will never be queried

Reply:
Thanks for testing! Sorry this was my mistake. This is very bad behavior, because in case of an issue with NSQ, the collector drops the events forever and never crashes. It should either retry the insert forever until it blows up or crash after the first failure (and restart). But given that this is just for Mini and the old collector was already behaving like this, let's keep as is.

Reply:
No apology needed - how you expected it to be is how it should be, I absolutely agree!